chat_v2_message_counter_sample/src/main/java/com/zendesk/chat_v2/sample/UnreadMessageCounter.java

```diff
@@ -0,0 +1,73 @@
+package com.zendesk.chat_v2.sample;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import zendesk.chat.Chat;
+import zendesk.chat.ChatLog;
+import zendesk.chat.ChatState;
+import zendesk.chat.ObservationScope;
+import zendesk.chat.Observer;
+
+class UnreadMessageCounter {
+
+    private final UnreadMessageCounterListener unreadMessageCounterListener;
+
+    private boolean shouldCount;
+    private String lastChatLogId;
+    private String lastReadChatLogId;
+    private List<ChatLog> unreadChatLogs = new ArrayList<>();
```

Comment:
Maybe this is a holdover from a previous implementation, but looks like we don't need this list? It seems to only be used in line 62 to pass the length so we can make it a local variable there